hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/table/HoodieFlinkTable.java

```
@@ -100,6 +100,9 @@ protected HoodieIndex getIndex(HoodieWriteConfig config, HoodieEngineContext con
   protected Option<HoodieTableMetadataWriter> getMetadataWriter(
       String triggeringInstantTimestamp,
       HoodieFailedWritesCleaningPolicy failedWritesCleaningPolicy) {
+    if (isMetadataTable()) {
+      return Option.empty();
```

we should throw an error from here? so it never reaches this sort of condition?